\documentclass[12pt]{article}
\usepackage[a4paper]{}
\usepackage{graphicx}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage[]{algorithm2e}
\usepackage{comment}
\usepackage{soul}
\usepackage{cancel}
\usepackage{multirow}
\usepackage{chngcntr}
\usepackage{enumitem}
\usepackage{booktabs}
\usepackage[table,xcdraw]{xcolor}
\counterwithin{table}{section}
\usepackage{subfigure}
\usepackage{array}
\usepackage{float}
\restylefloat{table}
\graphicspath{ {images/} }
%\topmargin 15mm
\numberwithin{figure}{section}
\providecommand{\keywords}[1]{\textbf{\textit{Keywords---}} #1}
\headheight 10pt
\oddsidemargin 0mm
%\evensidemargin -5mm
\textwidth 161mm
\textheight 210mm
\linespread{1.3}
\begin{document}
\thispagestyle{empty}
 \begin{center}
Major Project Proposal Report\\
\null
on\\
\null
\textbf{Automatic Generation of Medical Imaging Reports}\\
\null
Submitted by \\
\null
\textbf{Rahul Kumar}\\
\textbf{172IT014}\\
\textbf{M.Tech (IT)}\\
\null
Under the Guidance of\\
\textbf{Dr. Sownmya Kamath S.}  \\
\null
\textbf{Dept. of Information Technology,}\\
\textbf{NITK, Surathkal}\\
\null
\textbf{Date of Submission: 18-10-2018}\\
\begin{figure}[h] % b - bottom, t - top, h - here
 \centering{
 \includegraphics[scale=0.7]{logo.jpg}
 }
\end{figure}
\textbf{Department of Information Technology\\
National Institute of Technology Karnataka, Surathkal\\
2018-2019}\\
\end{center}
\clearpage
\thispagestyle{empty}
\begin{center}
\underline{\textbf{CERTIFICATE}} \\
\end{center}
\null
This is to certify that the major project entitled "Automatic Generation of Medical Imaging Reports" is a bonafide work carried out as part of the course Major Project (IT898), under my guidance by Rahul Kumar, student of III Semester M.Tech (IT) at the Department of Information Technology, National Institute of Technology Karnataka, Surathkal, during the academic semester Jul - Nov 2018, in partial fulfillment of the requirements for the award of the degree of Master of Technology in Information Technology, at NITK Surathkal. 
\\
\\
\\
\\
\\
\\
\begin{minipage}{0.4\textwidth}
\begin{flushleft}
Signature of the Student\\
Place: Mangalore\\ 
Date:28-09-2018\\
\end{flushleft}
\end{minipage}
\begin{minipage}{0.4\textwidth}
\begin{flushright}
Signature of the Instructor(s)\\
\end{flushright}
\end{minipage}
\\
\\
\\
\clearpage
\thispagestyle{empty}
\begin{center}
\underline{\textbf{DECLARATION}}\\
\end{center}
\null
I hereby declare that the Major Project entitled “Automatic Generation of Medical Imaging Reports” submitted as part of the partial course requirements for the course Major Project (IT898) , for the award of the degree of Master of Technology in Information Technology at NITK Surathkal during the Jul - Nov 2018 semester, has been carried out by me. I declare that the project has not formed the basis for the award of any degree, associate ship, fellowship or any other similar titles elsewhere.\\ 
\\
\\
\\
\\
\\
\\
Signature of the Student\\
Place: Mangalore\\
Date:28-09-2018\\
\clearpage
\pagenumbering{roman}
\section*{Abstract}
Apart from being an interesting computer-vision problem, manual analysis of radiology images also needs to be performed by experienced radiologist which makes this issue even more necessary to be addressed. Automatic description generation for radiology images can benefit radiologist with quick, localised and explanatory analysis. Expected analysis must contain keywords (Tags) that images relates with, observation for different body parts in the image ( Findings), and the diagnosis ( Impression ).
Our goal here is comprises of three tasks, to classify given X-ray image against list of medical abnormalities resulting in tagged images, to generate narration along with localising region for found abnormalities, and finally using hierarchical LSTM, generating long report description .
We are 

\\
Keywords - Machine learning, GAN, deep learning, neural nets, DCGAN.
\\
\clearpage
\tableofcontents
\clearpage
\listoffigures 
\listoftables
\clearpage
\pagenumbering{arabic}
\section{Introduction}
\paragraph{Medical images : }
Radiologist analyze and diagnose using medical images for various diseases on a daily basis. However analysis, reading and interpretation, and writing textual report demands skills such as knowledge of anatomy and basic details about the visual symptoms of the disease, knowledge of correlation with other diagnostic results like pathology result etc.
Writing diagnostic report for radiology image, even for experienced radiologists is a time consuming process. And this same task sometimes is required to be repeated for up to hundreds of different radiology images everyday.
\paragraph{Medical reporting task and parts of report : }
Medical reporting procedure for radiology images include radiologist analyzing various part of patient’s body and remarking various findings about those body parts. Sometimes different related or unrelated findings about different body parts can be observed in single images. Those findings and other observation then lead to a further consolidated overall diagnostic report about the patient known as Impression. In this process images needs to be given proper tags for future categorization and quick analysis purpose.
\paragraph{}
So our challenge is to provide solution for generating tags, findings and impression for a given radiology image. As the process of diagnostic, our challenge is can be broken down in three stages, starting with tags generation for a given image. We are solving this problem by classifying the images against various pre-known labels and thus solving it as a multi-label classification problem. Tags generated at this step are preserved for future query purpose as well as as input for further processing of images. 
\paragraph{}
We will be using Generative adversarial networks for multi-label classification. For our task, we setup a GAN for generating new sample data for N classes and then classifying it for N+1 classes, representing N real classes for N labels and one class for fake data.
Various regions of an image may contain different information for different body parts, which can be related sometimes or to be looked as unrelated multiple diagnostic many other times. This task includes localising the image regions and then generating narration for those regions. We are using a co-attention mechanism for analysing given images and the tags generated in multi-label classification process for the given image simultaneously and provide semantic information which can be used for further processing along with previously generated tags information.
\paragraph{Co-attention:}
Generating paragraph of description using hierarchical LSTM
Our third task is to generate descriptions for imaging report summarizing the findings for various regions of images. Description are usually long paragraphs having multiple lines. So this task presents challenges as generating long paragraphs is a nontrivial task. We will be using hierarchical LSTM\cite{1} with co-attention mechanism for generating findings for various regions and then combining them and further processing for description summary generation.
\paragraph{}
To summarize, our multi-task learning framework  will be generating tags, sentences for findings and multi-sentence textual description for given radiology images.


\clearpage


\section{Literature Survey}
\subsection{Background Study}
\paragraph{}
Generating text label for images has been addressed as a multi-label classification task in many previous research. My base paper used pre-trained vgg19 model to classify x-ray images for various class by extracting features from last convolution layer and last two fully connected layers of vgg-19 as multi-label classifier. vgg-19 is trained on a stack of convolutional-maxpooling layer followed by fully connected layers. Max-pooling causes loss of important spatial information and it makes the network ignore the relation between the part and the whole image.
\paragraph{}

\paragraph{}

\paragraph{}


\clearpage
\subsection{Outcome of Literature Survey}

\begin{table}[H]
\caption{Outcome of Literature Survey}
\label{Outcome of Literature Survey}
\centering
\resizebox{\textwidth}{!}{%
\begin{tabular}{|c|c|c|c|}
\hline
\rowcolor[HTML]{C0C0C0} 
\textbf{Authors} & \textbf{Method} & \textbf{Advantages} & \textbf{Limitations} \\ \hline
David T. Jones & \begin{tabular}[c]{@{}c@{}}Proteins Secondary\\ Structure Prediction\\ Based on Position\\ Specific Scoring\\ Matrix\end{tabular} & \begin{tabular}[c]{@{}c@{}}Using a new testing set\\ based on a set of 187\\ unique folds, and three\\ way cross-validation\\ based on structural\\ similarity.\end{tabular} & \begin{tabular}[c]{@{}c@{}}Takes too much time\\ for training the module\\ and predicting the\\ protein secondary
\\structure.\end{tabular} \\ \hline
\begin{tabular}[c]{@{}c@{}}Guoren Wang\\ et al.\end{tabular} & \begin{tabular}[c]{@{}c@{}}A protein secondary\\ structure prediction\\ framework based on\\ the Extreme\\ Learning Machine\end{tabular} & \begin{tabular}[c]{@{}c@{}}ELM is a new learning\\ method used in the training\\ of PSS framework which\\ consumes little training\\ time while achieving\\ high accuracy.\end{tabular} & \begin{tabular}[c]{@{}c@{}}Not much improvement\\ in the Q3 accuracy and\\ Segment Overlap (Sov)\\ score.\end{tabular} \\ \hline
\begin{tabular}[c]{@{}c@{}}Amouda\\ Venkatesan et al.\end{tabular} & \begin{tabular}[c]{@{}c@{}}Computational Approach\\ for Protein Structure\\ Prediction\end{tabular} & \begin{tabular}[c]{@{}c@{}}Contributes to the\\ prediction of a native-\\like structure by eliminating\\ the time constraint and\\ effort demand. Energy\\ of the predicted structure\\ is minimized to great\\ extent, which proves the\\ stability of protein. \end{tabular} & \begin{tabular}[c]{@{}c@{}}Tested only on two\\ peptides. Not focused on\\ the Q3 or Q8 accuracy\\ as well as not focused\\ on Sov accuracy.\end{tabular} \\ \hline
\begin{tabular}[c]{@{}c@{}}Matt Spencer\\ et al.\end{tabular} & \begin{tabular}[c]{@{}c@{}}A deep learning network\\ approach to ab initio\\ protein secondary\\ structure prediction\end{tabular} & \begin{tabular}[c]{@{}c@{}}Used multilayer deep\\ learning network with PSSM\\ and Atchley's factor.\\ Utilized a hybrid scoring\\ method that takes Q3\\ and Sov score into account\\ in an attempt to increase\\ the PSP accuracy.\end{tabular} & \begin{tabular}[c]{@{}c@{}}Not able to increase\\ the Q3 score\\ significantly.\end{tabular} \\ \hline
\end{tabular}
}
\end{table}
\paragraph{}
There are a significant amount of work already done in this field. But the problem of very high time consumption during the training of deep learning network \cite{3} or the problem of stagnating of the accuracy around 80 \% is still there \cite{6}. Extreme Learning Machine is able to reduce the training time significantly, but it did not able to improve the Q3 accuracy \cite{4}. In our work, we are using ELM to reduce the training time of our deep learning network framework and LSTM to help it to increase the Q3 accuracy.
 \subsection{Problem Statement}
 \paragraph{}
 Using GANs to improve multi-label classification efficiency for “Tag generation task” in Automatic Generation of natural Medical Imaging Reports
\subsection{Objectives}
\begin{itemize}
\item Set up PSI-BLAST and get position-specific scoring matrix.
\item Generation of Sequence profiles.
\item Training of deep learning network architectures.
\item Prediction of protein secondary structure and post processing of the results to optimize the results.
\end{itemize}
\clearpage
\section{Research Methodology and Materials}
\subsection{Datasets}
\paragraph{}
Our training dataset will be a collection of protein sequences from four datasets - CB6133, CB513, CASP10, and CASP11. CB6133 is a non-homologous protein structure and sequence dataset. This methodology uses the CB6133 dataset to train and test the deep neural models. The CB513, CASP10, CASP11 datasets are public benchmark dataset only used for testing. We will be evaluating the performance of our model based on Q3, Q8 accuracy and Sov score.
\subsection{Proposed Methodology}
Overview
We are proposing a multi-task framework to perform both tag prediction and description generation for given x-ray image. For a given input X-ray image, our framework will generate long natural reports covering diverse topic.  
\paragraph{}


\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{framwork.jpg}}
    \caption{Framework for automatic description generator}
    \label{fig:my_label}
\end{figure}
\subsubsection{Tag prediction}

We will use GANs tuned in for semi-supervised learning for multi-label classification of images and as well as for extracting feature vector to be used by co-attention mechanism later.
\paragraph{}
\subsubsection{Co-attention}
The previous step added tags to our information and now for utilizing both visual information of a region of the image and predicted tags. We are utilising co-attention mechanism to simultaneously attend to both modalities.
Joint context vector is generated using embedding vector for most likely tags, feature vector from MLC, combined with previous hidden state of Sentence LSTM generating soft semanntic attentions and soft visual attentions over given image features and tags. Further visual and semantic context vecotr, produced separately will be combined by concatenating.
Visual and semantic context vectors will be computed
\paragraph{}
\subsubsection{Sentence LSTM}
A single layer LSTM using joint context vector as its input generates topic vector for word LSTM, next level of hierarchical LSTM using topic generator combined with stop control mechanism.
\paragraph{}
\subsubsection{Word LSTM}
Taking the topic vector and START token as inital input 
\paragraph{}

\section{Conclusion and Future Work}
\paragraph{}
The Protein secondary structure prediction can be used in for the prediction of protein tertiary structure, for protein-protein interactions, and for drug discovery, depending upon the prediction efficiency. The more efficiently we predict the secondary structure, more efficiently we can predict the tertiary structures and get the functional information about those protein sequences. 
\paragraph{}
The future work is to find a set of features which will efficiently train model and will optimize the protein secondary structure prediction results.
\clearpage
\section{Time line of the project}
\begin{table}[htp]
\centering
\caption{Time line for completion of project}
\label{my-label}
\begin{tabular}{|
>{\columncolor[HTML]{EFEFEF}}l |l|l|l|l|l|l|}
\hline
\cellcolor[HTML]{CBCEFB}\textbf{Milestone} &                                      &                                &                                &                                &                                &                                    \\ \hline
Feasibility Study                          & \cellcolor[HTML]{C0C0C0}Jul-Sep 2018 &                                &                                &                                &                                &                                    \\ \hline
Objective 1                                &                                      & \cellcolor[HTML]{C0C0C0}Oct 18 &                                &                                &                                &                                    \\ \hline
Objective 2                                &                                      &                                & \cellcolor[HTML]{C0C0C0}Nov 18 &                                &                                &                                    \\ \hline
Objective 3                                &                                      &                                &                                & \cellcolor[HTML]{C0C0C0}Nov 18 - Dec 18 &                                &                                    \\ \hline
Objective 4                                &                                      &                                &                                &                                & \cellcolor[HTML]{C0C0C0}Dec 18 &                                    \\ \hline
Final Submission                           &                                      &                                &                                &                                &                                & \cellcolor[HTML]{C0C0C0}Jan 19 - May 19 \\ \hline
\end{tabular}
\end{table}
\clearpage
\addcontentsline{toc}{section}{References}
\begin{thebibliography}{}
\bibitem{1}
Smith, T.F. \& Waterman, M.S., “Identification of Common Molecular Subsequences”, Journal of Molecular Biology, Vol. 147, No. 1, Page No. 195-197, 1981.
\bibitem{2}
Altschul, S.F., madden, T.L., Schaffer, A.A., Zhang, J.H., Zhang, Z., Miller, W., \& Lipman, D.J., “Gapped BLAST and PSI-BLAST: a new generation of protein database search programs”, Nucleic Acids Research, Vol. 25, No. 17, Page No. 3389-3402, 1997.
\bibitem{8}
 Guang-Bin Huang, Qin-Yu Zhu, Chee-Kheong Siew, "Extreme learning machine: a new learning scheme of feedforward neural networks", IEEE International Joint Conference on Neural Networks, (IJCNN'2004), Vol. 2, Page No. 985-990, 2004.
 \bibitem{9}
 Guang-Bin Huang, Qin-Yu Zhu, Chee-Kheong Siew, "Extreme learning machine: theory and applications", Neurocomputing, Vol. 70, N0. 1-3, Page No. 489-501, 2006.
\bibitem{3}
David T. Jones, “Protein secondary structure prediction based on position-specific scoring matrices”, Journal of Molecular Biology, Vol. 292, No. 2, Page No. 195-202, 1999.
\bibitem{4}
Guoren Wang, Yi Zhao, Di Wang “A protein secondary structure prediction framework based on the Extreme Learning Machine”, Neurocomputing, Vol. 72, No. 1-3, Page No. 262-268, 2008.
\bibitem{5}
Amouda Venkatesan, Jeyakodi Gopal, Manimozhi Candavelou, Sowjanya Gollapalli, "Computational Approach for Protein Structure Prediction", Healthcare Informatics Research, Vol. 19, No. 2, Page No. 137-147, 2013
\bibitem{6}
Matt Spencer, Jesse Eickholt, and Jianlin Cheng, “A Deep learning Network Approach to ab initio Protein Secondary Structure Prediction”, IEEE/ACM Transactions on Computational Biology and Bioinformatics, Vol. 12, No. 1, Page No. 103-112, 2015 
\bibitem{7}
Yanbu Guo, Bingyi Wang, Weihua Li, Bei Yang, "Protein secondary structure prediction improved by recurrent neural networks integrated with 2-dimensional convolutional neural networks", Journal of Bioinformatics and Computational Biology, 2018
\end{thebibliography}

\end{document}